js/graviex.js

```diff
@@ -0,0 +1,987 @@
+        //     "blocks": 260722,
+        //     "block_time": "2022-01-31 14:10:08",
+        //     "wstatus": "on",
+        //     "low": "0.0000005",
+        //     "high": "0.00000058",
+        //     "last": "0.00000058",
+        //     "open": "0.000000510",
+        //     "volume": "70673.0412",
+        //     "volume2": "0.03837782973",
+        //     "sell": "0.00000057",
+        //     "buy": "0.00000055",
+        //     "at": 1643628325,
+        //   },
+        // },
+        // const data = response;
+        // let timestamp = data['at'];
+        // let tickers = data['ticker'];
+        const ids = Object.keys (response);
+        const result = {};
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const market = this.safeMarket (id);
+            const symbol = market['symbol'];
+            result[symbol] = this.parseTicker (response[id], market);
+        }
+        return result;
+    }
+
+    parseTrade (trade, market = undefined) {
+        // {
+        //     "id": 14473952,
+        //     "at": 1643681026,
+        //     "price": "0.00000056",
+        //     "volume": "0.2",
+        //     "funds": "0.000000112",
+        //     "market": "giobtc",
+        //     "created_at": "2022-02-01T05:03:46+03:00",
+        //     "side": "sell"
+        // },
+        const price = this.safeString2 (trade, 'p', 'price');
+        const amount = this.safeString2 (trade, 'volume', 'amount');
+        const marketId = this.safeString (trade, 'market');
+        const symbol = this.safeSymbol (marketId, market);
+        const timestamp = this.safeTimestamp (trade, 'at');
+        let id = this.safeString2 (trade, 't', 'a');
+        id = this.safeString2 (trade, 'id', 'tid', id);
+        const side = this.safeValue (trade, 'side');
+        let takerOrMaker = undefined;
+        const orderId = this.safeString (trade, 'id');
+        if ('side' in trade) {
+            if (side === 'buy') {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': orderId,
+            'type': undefined,
+            'side': side,
+            'takerOrMaker': takerOrMaker,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        const defaultType = 'spot';
+        const type = this.safeString (params, 'type', defaultType);
```

Comment:
let's add `handleMarketTypeAndParams` here

Reply:
Okay. Fixed in the latest commit https://github.com/ccxt/ccxt/commit/81c8152ac30b15da7aad6a3f117da386d51efd62